apps/application-system/api/src/app/modules/payment/payment.controller.ts

```diff
@@ -140,11 +145,28 @@ export class PaymentController {
       )
     }
 
+    const application = await this.applicationService.findOneById(applicationId)
+
+    let applicationSlug
+    if (application?.typeId) {
+      applicationSlug = getSlugFromType(application.typeId)
+    } else {
+      throw new NotFoundException(
+        `application type id was not found for application id ${applicationId}`,
+      )
+    }
+
+    const callbackUrl = `${environment.templateApi.clientLocationOrigin}/${applicationSlug}/${applicationId}?done`
+
     return {
       // TODO: maybe treat the case where no payment was found differently?
       // not sure how/if that case would/could come up.
       fulfilled: payment.fulfilled || false,
-      paymentUrl: this.paymentService.makePaymentUrl(payment.user4),
+      paymentUrl: this.paymentService.makeDelegationPaymentUrl(
+        payment.user4,
+        user.sub ?? user.nationalId,
```

Comment:
No, it was just a way to handle optional field.  Since we're technically not allowed to use nationalId there I made a different null check and removed the fallback.